analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -0,0 +1,457 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+
+namespace Tests.Diagnostics
+{
+    class CollectionTests
+    {
+        private IEnumerable<int> items = new List<int> { 1, 2, 3 };
+        private IDictionary<int, int> dictionaryItems = new Dictionary<int, int>();
+
+        private static bool Predicate(int i) => true;
+        private static void Action(int i) { }
+        private List<int> ListFactory() => new List<int>();
+        private HashSet<int> SetFactory() => new HashSet<int>();
+        private Queue<int> QueueFactory() => new Queue<int>();
+        private Stack<int> StackFactory() => new Stack<int>();
+        private ObservableCollection<int> ObservableCollectionFactory() => new ObservableCollection<int>();
+        private int[] ArrayFactory() => new int[0];
+        private Dictionary<int, int> DictionaryFactory() => new Dictionary<int, int>();
+
+        public void DefaultConstructor()
```

Comment:
This file is too difficult to review. I can't tell what changes are improvements and what changes are mistakes lost during the migration.

The intention of scaffolding was to have a file as similar as possible to the old one (except namespace, FIXMEs, C# 8 extraction and maybe removal of irrelevant blocks). 
And than somewhere else, like the implementation PR, have a clearly distinguished improvements visible.